### DISCUSSION TITLE
/hyndsight/tsibbles/

### DISCUSSION THREAD

**Discussion by robjhyndman** [robjhyndman/robjhyndman.com#312] [Announcements]
Tidy time series data using tsibbles 
 https://robjhyndman.com/hyndsight/tsibbles/

**Comment by robjhyndman** [comment]
```header
Originally posted by Steff on 2019-09-02 08:38:17
````
Hi Rob,

really intresting and helpful post. Just one question, if i convert my
data from raw to tibble with the command &quot;as_tsibble&quot; do the function
need a frequency like the &quot;ts&quot; command?\
Because i\'ve monthly data but str output this:

&quot; ..\$ .rows:List of 1\
.. ..\$ : int 1 2 3 4 5 6 7 8 9 10 \...\
- attr(\*, &quot;index&quot;)= chr &quot;Date&quot;\
..- attr(\*, &quot;ordered&quot;)= logi TRUE\
- attr(\*, &quot;index2&quot;)= chr &quot;Date&quot;\
- attr(\*, &quot;interval&quot;)=List of 12\
..\$ year : num 0\
..\$ quarter : num 0\
..\$ month : num 0\
..\$ week : num 0\
..\$ day : num 1\
..\$ hour : num 0\
..\$ minute : num 0\
..\$ second : num 0\
..\$ millisecond: num 0\
..\$ microsecond: num 0\
..\$ nanosecond : num 0\
..\$ unit : num 0\
..- attr(\*, &quot;class&quot;)= chr &quot;interval&quot;&quot;

Thanks and have a good day!

BR from Germany\
Steff

**Reply by robjhyndman** [reply]
```header
Originally posted on 2019-09-03 07:42:06
````
I don\'t know what you mean by &quot;raw&quot;. as_tsibble() is for converting
tibbles, data frames or ts objects to tsibble class. If the data is in a
data frame or tibble, then you will need to specify the index containing
the dates or times. See the extensive package documentation for
examples:
[https://tsibble.tidyverts.org/](https://tsibble.tidyverts.org/)

**Reply by robjhyndman** [reply]
```header
Originally posted by Mitchell O'Hara-Wild on 2019-09-24 00:58:50
````
Hi Steff,\
To convert data from a data.frame to a tsibble you will need to convert
the time index (your dates) into the appropriate class. In this case you
should use \`yearmonth()\` to convert your dates (daily time
measurement) into year months (monthly time measurement). If this is
done before creating a tsibble (with \`as_tsibble()\`), then the monthly
frequency will automatically be detected. Unlike \`ts\` objects, you do
not need to specify the frequency with \`tsibble\`.

**Comment by robjhyndman** [comment]
```header
Originally posted by Michael He on 2019-09-11 19:47:32
````
Thanks a lot for this data!

**Comment by MikNech7777** [comment]
Hi Rob,

It is not clear if we can apply 'difference' in case a tsibble contains several time series. Can we get difference for all series in a column? For example, if I have several stocks by their tickers in three columns like 'Date', 'Ticker', 'Price', can I get a column 'Diff' with consequent differences in price for all tickers? Sorry if the question is trivial ...

**Reply by MikNech7777** [reply]
OK. It looks like group_by solves the question:

zz |> group_by(Ticker) |> mutate(diff = difference(Price, order_by = Date))
